src/vma/lwip/tcp_out.c

```diff
@@ -1030,7 +1030,7 @@ tcp_output(struct tcp_pcb *pcb)
 
   while (seg){
     /* Split the segment in case of a small window */
-    if (( pcb->flags & (TF_NODELAY | TF_INFR)) && (wnd) && ((seg->len + seg->seqno - pcb->lastack) > wnd)) {
+    if (((seg->len + seg->seqno - pcb->lastack) > wnd) && (wnd) && (NULL == pcb->unacked)) {
```

Comment:
Please consider changing a order of conditionals to following
`if ((NULL == pcb->unacked) && (wnd) && (seg->len + seg->seqno - pcb->lastack) > wnd))`

Reply:
:+1: